src/app/Dashboard/Charts/ChartCard.tsx

```diff
@@ -0,0 +1,367 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { CreateRecordingProps } from '@app/CreateRecording/CreateRecording';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { FeatureLevel } from '@app/Shared/Services/Settings.service';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  Bullseye,
+  Button,
+  CardActions,
+  CardBody,
+  CardHeader,
+  CardTitle,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Label,
+  Title,
+} from '@patternfly/react-core';
+import { DataSourceIcon, ExternalLinkAltIcon, SyncAltIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router-dom';
+import { interval } from 'rxjs';
+import { DashboardCardDescriptor, DashboardCardProps, DashboardCardSizes } from '../Dashboard';
+import { DashboardCard } from '../DashboardCard';
+import { ChartContext } from './ChartContext';
+import { RECORDING_NAME } from './ChartController';
+
+export interface ChartCardProps extends DashboardCardProps {
+  theme: string;
+  chartKind: string;
+  duration: number;
+  period: number;
+}
+
+// TODO these need to be localized
+export enum ChartKind {
+  'Core Count' = 1,
+  'Thread Count' = 2,
+  'CPU Load' = 3,
+  'Heap Usage' = 4,
+  'Memory Usage' = 5,
+  'Total Memory' = 6,
+  'Recording Start Time' = 7,
+  'Recording Duration' = 8,
+  'Classloading Statistics' = 9,
+  'Metaspace Summary' = 10,
+  'Network Utilization' = 11,
+  'Metaspace GC Threshold' = 12,
+  'Thread Statistics' = 13,
+  'Exception Statistics' = 14,
+  'Thread Context Switch Rate' = 15,
+  'Compiler Statistics' = 16,
+
+  'Safepoint Duration' = 18,
+  'File I/O' = 19,
+  'Compiler Total Time' = 20,
+
+  'Compiler Peak Time' = 24,
+
+  'Object Allocation Sample' = 38,
+}
+
+export function kindToId(kind: string): number {
+  return ChartKind[kind];
+}
+
+export const ChartCard: React.FC<ChartCardProps> = (props) => {
+  const [t] = useTranslation();
+  const serviceContext = React.useContext(ServiceContext);
+  const controllerContext = React.useContext(ChartContext);
+  const history = useHistory();
+  const addSubscription = useSubscriptions();
+  const [key, setKey] = React.useState(Math.floor(Math.random()));
+  const [hasRecording, setHasRecording] = React.useState(false);
+  const [chartSrc, setChartSrc] = React.useState('');
+  const [dashboardUrl, setDashboardUrl] = React.useState('');
+
+  const updateKey = React.useCallback(() => {
+    setKey((prev) => {
+      let next = prev + 1;
+      if (next >= 10) {
+        next = 0;
+      }
+      return next;
+    });
+  }, [setKey]);
+
+  React.useEffect(() => {
+    addSubscription(serviceContext.target.target().subscribe((_) => updateKey()));
```

Comment:
Any bugs that cause us to force listen and force iframe reload since we already listen to this in ChartController?